extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -13,3 +13,127 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Service for recaptcha components
+ * interacting with the 3p recaptcha bootstrap iframe
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {dev} from '../../../src/log';
+import {getIframe} from '../../../src/3p-frame';
+import {getService, registerServiceBuilder} from '../../../src/service';
+import {listenFor} from '../../../src/iframe-helper';
+import {removeElement} from '../../../src/dom';
+
+export class AmpRecaptchaService {
+
+  /**
+   * @param {!Window} window
+   */
+  constructor(window) {
+
+    /** @private {!Window} */
+    this.win_ = window;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @const @private {!Array} */
+    this.registeredElements_ = [];
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {!Deferred} */
+    this.willBeReady_ = new Deferred();
+  }
+
+  /**
+   * Function to register as a dependant of the AmpRecaptcha serivce.
+   * Used to create/destroy recaptcha boostrap iframe.
+   * @param {!AMP.BaseElement} elementImpl
+   */
+  register(elementImpl) {
+    this.registeredElements_.push(elementImpl);
+    if (!this.iframe_) {
+      return this.initialize_(elementImpl);
+    }
+    return Promise.resolve();
```

Comment:
There's a gap between `this.iframe_` is created, and iframe is ready. You want to always return the iframe ready promise instead of `Promise.resolve()`